extensions/common/context.cc

```diff
@@ -106,6 +108,16 @@ void getDestinationService(const std::string& dest_namespace,
                              ->toString()
                        : "unknown";
 
+  // override the cluster name if this is being sent to the
+  // blackhole or passthrough cluster
+  std::string route_name;
+  getValue({"route_name"}, &route_name);
+  if (route_name == kBlackHoleRouteName) {
+      cluster_name = kBlackHoleCluster;
+  } else if (route_name == kPassThroughRouteName) {
+      cluster_name = kPassThroughCluster;
```

Comment:
Let's get this in for now and unblock users.

We can implement the generic metadata approach as a follow-on.

Reply:
I'll write an issue for the metadata approach and put it for 1.6 milestone.